packages/api-client/src/server/tally.ts

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable max-classes-per-file */
+import { BaseQueryRequest, BaseResponse } from '../models/base';
+import { CiphertextElectionContext } from '../models/election';
+import {
+    DecryptionShareResponse,
+    CiphertextTally,
+    CiphertextTallyDecryptionShare,
+    CiphertextTallyQueryResponse,
+    DecryptionShareRequest,
+    DecryptTallyRequest,
+    DecryptTallyShareRequest,
+    PlaintextTally,
+    PlaintextTallyQueryResponse,
+} from '../models/tally';
+import { get, post } from '../utils/http';
+
+export const decryptShareTally = async (
+    election_id: string,
+    tally_name: string
+): Promise<CiphertextTallyDecryptionShare[] | undefined> => {
+    const path = `${process.env.REACT_APP_GUARDIAN_SERVICE}tally/decrypt-share?election_id=${election_id}&tally_name=${tally_name}`;
+    const response = await get<{ resp: DecryptionShareResponse }>(path);
+    return response.parsedBody?.resp.shares;
+};
+
+export const decryptSharePostTally = async (
+    guardian_id: string,
+    encrypted_tally: CiphertextTally,
+    context: CiphertextElectionContext
+): Promise<CiphertextTallyDecryptionShare[] | undefined> => {
+    const data: DecryptTallyShareRequest = {
+        guardian_id,
+        encrypted_tally,
+        context,
+    };
+
+    const path = `${process.env.REACT_APP_GUARDIAN_SERVICE}tally/decrypt-share`;
+    const response = await post<{ resp: DecryptionShareResponse }>(path, data);
+    return response.parsedBody?.resp.shares;
+};
+
+export const getTallyDecrypt = async (
+    election_id: string,
+    tally_name: string,
+    guardian_id: string
+): Promise<CiphertextTallyDecryptionShare[] | undefined> => {
+    const path = `${process.env.REACT_APP_MEDIATOR_SERVICE}tally/decrypt?election_id=${election_id}&tally_name=${tally_name}&guardian_id=${guardian_id}`;
+    const response = await get<{ resp: DecryptionShareResponse }>(path);
+    return response.parsedBody?.resp.shares;
+};
+
+export const postShareTally = async (
+    share: CiphertextTallyDecryptionShare
+): Promise<boolean | undefined> => {
+    const data: DecryptionShareRequest = {
+        share,
+    };
+    const path = `${process.env.REACT_APP_MEDIATOR_SERVICE}tally/submit-share`;
+    const response = await post<{ resp: BaseResponse }>(path, data);
+    return response.parsedBody?.resp.is_success();
+};
+
+export const findTallyDecrypt = async (
+    tally_name: string,
+    skip: number,
+    limit: number
+): Promise<CiphertextTallyDecryptionShare[] | undefined> => {
+    const data: BaseQueryRequest = {
+        filter: {},
+    };
+    const path = `${process.env.REACT_APP_MEDIATOR_SERVICE}tally/find?tally_name=${tally_name}&skip=${skip}&limit=${limit}`;
+    const response = await post<{ resp: DecryptionShareResponse }>(path, data);
+    return response.parsedBody?.resp.shares;
```

Comment:
Long term maybe we can build something to extract the parsed body for all the responses. Save on repeated code.

Reply:
This is something that we can look into but all of the parsed bodies are different.